### ISSUE TITLE
Running into session limit?

### ISSUE THREAD

**Issue by apoliakov** [Paradigm4/shim#85]
Hey @rvernica here's an interesting issue we bumped into today. Running this:
```python
from scidbpy import connect
import gc
for i in range(1,100) :
    db=connect()
    print(db.iquery("list()", fetch=True).shape)
    del db
    gc.collect()
```

After about 50 (which is the default max shim number of sessions) it gets a 503 error. But I thought that when we clean up the db object, we disconnect, no?

**Comment by rvernica** [comment]
Interestin, SciDB-Py is supposed to issue `release_session`, see [here](https://github.com/Paradigm4/SciDB-Py/blob/master/scidbpy/db.py#L131)

**Comment by rvernica** [comment]
Fixed in [c8fa18f](https://github.com/Paradigm4/SciDB-Py/commit/c8fa18f3fd54d4aed803a78b67577b40deed6a28) A few notes:
* This was a SciDB-Py issue
* SciDB-Py was releasing the session when the Python interpreter exit even before this fix
* The `finalize` call was getting a reference to the DB object and hence it was never freed. This has been fixed.
* The above code works now. It was added as a test in SciDB-Py. The code might also work without the `collect` call, as Python my trigger the garbage collector before Shim exhausts the connection pool.
